pkg/virt-controller/watch/vm/vm.go

```diff
@@ -2936,6 +2936,11 @@ func (c *Controller) addRestartRequiredIfNeeded(lastSeenVMSpec *virtv1.VirtualMa
 		}
 	}
 
+	if !netvmispec.ShouldChangeRequireRestart(currentVM, vmi) {
+		lastSeenVM.Spec.Template.Spec.Domain.Devices.Interfaces = currentVM.Spec.Template.Spec.Domain.Devices.Interfaces
+		lastSeenVM.Spec.Template.Spec.Networks = currentVM.Spec.Template.Spec.Networks
+	}
+
 	if !equality.Semantic.DeepEqual(lastSeenVM.Spec.Template.Spec, currentVM.Spec.Template.Spec) {
```

Comment:
Based on this, if the order is changed + a non-reset change ( e.g. `state` or hotplug) on the new VM, `!netvmispec.ShouldChangeRequireRestart` will resolve to `false` but here we will resolve to `true` and therefore mark as restart-required which seems wrong to me.

We need the e2e test to cover this, because I do not see how this can be covered in unit tests at the moment.

Reply:
Good catch!
IIUC, SIG storage have the same problem @alicefr.

A test was added on our side to check that we don't consider the interfaces and networks ordering.

> We need the e2e test to cover this, because I do not see how this can be covered in unit tests at the moment.

This is still an open point.

Reply:
Looking at this again, there is no actual problem except a coverage one which has been taken care of by unit testing the required-restart logic. If a reset is not required, the network spec is synced so it will not effect the equality check here.

The e2e are needed only if we cannot unit-test the overall logic here.